File: app/src/main/java/com/example/jarsical/fragments/library/libraryFragment.java
package com.example.jarsical.fragments.library;

import static android.content.Context.MODE_PRIVATE;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.jarsical.R;
import com.example.jarsical.Song;
import com.example.jarsical.SongCollection;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.ArrayList;
import java.util.List;

public class LibraryFragment extends Fragment {
    RecyclerView recyclerView;
    List<Song> song;
    SharedPreferences sharedPreferences;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_library, container, false);
        recyclerView = view.findViewById(R.id.recyclerViewLibrary);
        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));

        sharedPreferences = getContext().getSharedPreferences("playlist", MODE_PRIVATE);
        String albums = sharedPreferences.getString("playlist","");
        if(!albums.equals("")){
            TypeToken<ArrayList<Song>> token = new TypeToken<ArrayList<Song>>(){};
            Gson gson = new Gson();
            song = gson.fromJson(albums,token.getType());
        }

        recyclerView.setAdapter(new LibraryAdapter(song));
        return view;
    }
}

File: app/src/main/java/com/example/jarsical/fragments/library/libraryFragment.kt
package com.example.jarsical.fragments.library

import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.jarsical.R
import com.example.jarsical.Song
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import kotlinx.android.synthetic.main.fragment_library.view.*

class libraryFragment : Fragment(){
    var sharedPreferences = SharedPreferences
    var favList: ArrayList<Song> = ArrayList()


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?

    ): View? {
        // Inflate the layout for this fragment
        val view =  inflater.inflate(R.layout.fragment_library, container, false)

        var albums = sharedPreferences.getString("favList","")

        if(!albums.equals("")){
            var token = object : TypeToken<ArrayList<Song>>(){}.type
            var gson = Gson()
            favList = gson.fromJson(albums,token)
            Log.d("libraryAdapter","IT WORKS JARVIS")
        }

        //recyclerview
        val adapter = libraryAdapter(favList)
        val recyclerView = view.recyclerViewLibrary
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(requireContext())

        return view

    }
}